homeassistant/components/mqtt/fan.py

```diff
@@ -353,16 +575,31 @@ async def async_set_speed(self, speed: str) -> None:
             mqtt_payload = self._payload["SPEED_HIGH"]
         elif speed == SPEED_OFF:
             mqtt_payload = self._payload["SPEED_OFF"]
+        elif (
+            self._feature_preset_mode and speed in self._config[CONF_PRESET_MODES_LIST]
+        ):
+            percentage = ordered_list_item_to_percentage(
+                self._config[CONF_PRESET_MODES_LIST], speed
+            )
+            mqtt_payload = percentage_to_ordered_list_item(
+                self._speed_list_without_preset_modes, percentage
+            )
+        #        elif self._feature_percentage:
+        #            percentage = int(speed)
+        #            if percentage:
+        #                mqtt_payload = percentage_to_ordered_list_item(
+        #                    self._speed_list_without_preset_modes, percentage
+        #                )
```

Comment:
Why is there commented out code here?

Reply:
This code is not hit by any scenario. It was not in the previous code base. The comment was removed.

Reply:
The code is not hit in any use case. It was not in the original code base. I removed the commented out code.